### PR TITLE
React to aspnet/Mvc#4834

### PR DESCRIPTION
**Pull request by dougbu**
- JSON now serialized with camel case
- another part of fix for aspnet/Mvc#4283

### CODE REVIEW

File: samples/MusicStore.Spa/Infrastructure/SmartJsonResult.cs [hunk 1]
```diff
@@ -29,8 +28,8 @@ public override Task ExecuteResultAsync(ActionContext context)
             //    context.HttpContext.Response.ContentType = "application/json";
             //    context.HttpContext.Response.ContentEncoding = Encoding.UTF8;
             //}
-            
-            return context.HttpContext.Response.WriteAsync(JsonConvert.SerializeObject(Data, Settings ?? new JsonSerializerSettings()));
+            var settings = Settings ?? JsonSerializerSettingsProvider.CreateSerializerSettings();
```
Comment (dougbu) [inline samples/MusicStore.Spa/Infrastructure/SmartJsonResult.cs]: We shouldn't see `new JsonSerializerSettings()` outside MVC unit tests and maybe not there.

**Comment by dougbu** [comment]
/to @javiercn since this is part of a bundle

**Comment by javiercn** [comment]
:shipit:

**Comment by dougbu** [comment]
6c53784

**Pull request merged**